### PR TITLE
Add new fields on Reading

### LINKED ISSUE #277: Implement example of MediaType assignment on binary device readings
The `MediaType` property has been added to go-mod-core-contracts value descriptor. Update device-simple01 example to assign MediaType to binary Event readings; e.g., mediaType='PNG'.

First determine if MediaType could be assigned in response to DeviceInitialize (new ProtocolDriver callback described in issue #276 ), or by directly mirroring the MediaType property (after adding to CommandValue structure).

### LINKED ISSUE #465: Handle the new Reading fileds- ValueType, MediaType, and FloatEncoding
See the related issue and PR

https://github.com/edgexfoundry/go-mod-core-contracts/issues/193
https://github.com/edgexfoundry/go-mod-core-contracts/pull/223

### PR DESCRIPTION
**Pull request by hahattan**
Fixes #465 
Fixes #277 

Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-CommitMessages
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by hahattan** [comment]
I think we could solve this together with #468 (or after), in this case we don't need the TypeToString() function in the commit.

File: pkg/models/commandvalue.go [hunk 1]
```diff
@@ -359,6 +359,40 @@ func (cv *CommandValue) ValueToString(encoding ...string) (str string) {
 	return
 }
 
+//TypeToString returns corresponding string representation of the ValueType.
+func (cv *CommandValue) TypeToString() string {
```
Comment (cloudxxx8) [inline pkg/models/commandvalue.go:363]: rename to ValueTypeToString will be clearer

File: pkg/models/commandvalue.go [hunk 2]
```diff
@@ -359,6 +359,40 @@ func (cv *CommandValue) ValueToString(encoding ...string) (str string) {
 	return
 }
 
+//TypeToString returns corresponding string representation of the ValueType.
```
Comment (cloudxxx8) [inline pkg/models/commandvalue.go:362]: please also update the go doc

File: pkg/models/commandvalue.go [hunk 3]
```diff
@@ -359,6 +359,40 @@ func (cv *CommandValue) ValueToString(encoding ...string) (str string) {
...
+func (cv *CommandValue) ValueTypeToString() string {
+	switch cv.Type {
+	case Bool:
+		return "Bool"
```
Comment (cloudxxx8) [inline pkg/models/commandvalue.go:366]: Don't use the hard code string, and please return the constant value in contract instead
https://github.com/iain-anderson/go-mod-core-contracts/blob/505d356ef65bdd8749da1347940cb4bba2b9fade/models/reading.go#L23-L35

**Comment by cloudxxx8** [comment]
> I think we could solve this together with #468 (or after), in that case we don't need the TypeToString() function in the commit.

There will be lots of change, so please open another PR to resolve it later.

**Comment by hahattan** [comment]
update `mediaType` attribute in Device Profile, this should  fix #277

**Pull request merged**